test/devices/temperature_control_test.py

```diff
@@ -33,35 +37,34 @@ def test_temperature_control_create():
     assert temperature_controller.temperature_last == 0
 
 
-def test_temperature_control_update():
+def test_temperature_control_at_temperature():
     """
-    The function to test updating the temperature controller
+    The function to test the temperature controller's at_temperature function
     """
     temperature_controller = create_temperature_controller()
 
-    temperature_controller.update()
-    time.sleep(1)
-    temperature_controller.update()
+    temperature_controller.temperature_probe.sensor.set_temperature(30.5)
+    assert temperature_controller.at_temperature() is True
 
+    temperature_controller.temperature_probe.sensor.set_temperature(30.6)
+    assert temperature_controller.at_temperature() is False
 
-def test_temperature_control_at_temperature():
-    """
-    The function to test the at_temperature function
-    """
-    temperature_controller = create_temperature_controller()
-
-    temperature_controller.at_temperature()
+    temperature_controller.temperature_probe.sensor.set_temperature(29.5)
+    assert temperature_controller.at_temperature() is True
 
+    temperature_controller.temperature_probe.sensor.set_temperature(29.4)
     assert temperature_controller.at_temperature() is False
 
 
-def test_temperature_control_last_temperature():
+def test_temperature_control_update():
     """
-    The function to get the last temperature from the temperature controller
+    The function to test updating the temperature controller
     """
     temperature_controller = create_temperature_controller()
 
-    assert temperature_controller.get_last_temperature() == 0
+    temperature_controller.update()
+    time.sleep(1)
+    temperature_controller.update()
```

Comment:
This will run the code, but there is no assert to confirm an expected result.

Reply:
This test is outdated and should have been removed now that the new tests exist and validate the results of the update function. This test is now removed.